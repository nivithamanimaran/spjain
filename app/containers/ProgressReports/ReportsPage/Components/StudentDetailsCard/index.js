import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import StudentProfileCardHeader from './Components/StudentProfileCardHeader';
import { useTheme } from '@material-ui/core';
import StudentProfileCardContent from './Components/StudentProfileCardContent';

const useStyles = makeStyles(theme => ({
  root: {
    // maxWidth: 345,
    border: `5px solid ${theme.palette.KenColors.kenWhite}`,
    borderRadius: '10px',
  },
  cardContent: {
    backgroundColor: theme.palette.KenColors.neutral11,
  },
  actionIcon: {
    cursor: 'pointer',
  },
  cardTitle: {
    fontWeight: 'bold',
    color: theme.palette.KenColors.tertiaryBlue501,
  },
}));

export default function StudentDetailsCard(props) {
  const {
    data,
    students,
    setStudentInfo,
    studentInfo,
    disableSearch = false,
  } = props;
  const classes = useStyles();
  const theme = useTheme();

  return (
    <Card
      className={classes.root}
      elevation={0}
      data-testid="student-details-card"
    >
      {!disableSearch ? (
        <StudentProfileCardHeader
          data={students}
          setStudentInfo={setStudentInfo}
          studentInfo={studentInfo}
        />
      ) : null}
      <CardContent
        className={classes.cardContent}
        data-testid="student-details-card-content"
      >
        <StudentProfileCardContent data={data} />
      </CardContent>
    </Card>
  );
}
