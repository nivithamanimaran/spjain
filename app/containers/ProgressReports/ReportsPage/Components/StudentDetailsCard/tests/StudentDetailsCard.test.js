import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import StudentDetailsCard from '../index';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

const data = {
  name: 'Fake Name',
  avatarImageUrl: 'Fake Image',
  details: { class: 'Montessori' },
};
describe('Student-Details-Card-Page', () => {
  it('Student Details Card renders correctly with data', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <StudentDetailsCard data={data} />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <StudentDetailsCard data={data} />
        </Router>
      </ThemeProvider>
    );

    const card = getByTestId('student-profile-card-header');
    expect(card).toBeTruthy();
    expect(card.textContent).toBe(data.name[0] + data.name);
    expect(getByTestId('student-details-card')).toMatchSnapshot();

    const cardContent = getByTestId('student-profile-card-content');
    expect(cardContent).toBeTruthy();
    expect(cardContent.textContent).toBe('ClassMontessori');
  });
  it('Student Details Card renders correctly without data', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <StudentDetailsCard />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <StudentDetailsCard />
        </Router>
      </ThemeProvider>
    );

    const cardHeader = getByTestId('student-profile-card-header');
    expect(cardHeader).toBeTruthy();
    expect(cardHeader.textContent).toBe('');

    const cardContent = getByTestId('student-profile-card-content');
    expect(cardContent).toBeTruthy();
    expect(cardContent.textContent).toBe('');
  });
});
