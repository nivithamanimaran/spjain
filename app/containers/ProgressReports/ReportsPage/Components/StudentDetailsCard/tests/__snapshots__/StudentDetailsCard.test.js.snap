// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student-Details-Card-Page Student Details Card renders correctly with data 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation0 MuiPaper-rounded"
  data-testid="student-details-card"
>
  <div
    class="MuiBox-root MuiBox-root-19"
    data-testid="student-profile-card-header"
  >
    <div
      class="MuiBox-root MuiBox-root-20"
      style="justify-content: center;"
    >
      <div
        class="MuiBox-root MuiBox-root-21"
        data-testid="student-profile-avatar"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-5 MuiAvatar-colorDefault"
          style="color: rgb(0, 33, 141); background-color: rgb(223, 225, 230);"
        >
          F
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-22 makeStyles-text-6"
        data-testid="student-profile-name"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-displayBlock"
            style="font-weight: 1000; color: rgb(0, 33, 141);"
          >
            Fake Name
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiCardContent-root makeStyles-cardContent-2"
    data-testid="student-details-card-content"
  >
    <div
      class="MuiBox-root MuiBox-root-23 makeStyles-box-14"
      data-testid="student-profile-card-content"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiBox-root MuiBox-root-24 makeStyles-innerBox-15"
          >
            <div
              data-testid="student-profile-card-content-label"
            >
              <p
                class="MuiTypography-root makeStyles-headings-13 MuiTypography-body2"
              >
                Class
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-25"
            >
              <div
                data-testid="student-profile-card-content-value"
              >
                <p
                  class="MuiTypography-root makeStyles-text-12 MuiTypography-body2 MuiTypography-colorPrimary"
                >
                  Montessori
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
